tests/suites/test_suite_ecdsa.function

```diff
@@ -212,25 +212,39 @@ void ecdsa_prim_random( int id )
     mbedtls_mpi d, r, s;
     mbedtls_test_rnd_pseudo_info rnd_info;
     unsigned char buf[MBEDTLS_MD_MAX_SIZE];
+    int test_runs = 2;
 
     mbedtls_ecp_group_init( &grp );
     mbedtls_ecp_point_init( &Q );
     mbedtls_mpi_init( &d ); mbedtls_mpi_init( &r ); mbedtls_mpi_init( &s );
     memset( &rnd_info, 0x00, sizeof( mbedtls_test_rnd_pseudo_info ) );
     memset( buf, 0, sizeof( buf ) );
 
-    /* prepare material for signature */
-    TEST_ASSERT( mbedtls_test_rnd_pseudo_rand( &rnd_info,
-                                               buf, sizeof( buf ) ) == 0 );
-    TEST_ASSERT( mbedtls_ecp_group_load( &grp, id ) == 0 );
-    TEST_ASSERT( mbedtls_ecp_gen_keypair( &grp, &d, &Q,
-                                          &mbedtls_test_rnd_pseudo_rand,
-                                          &rnd_info ) == 0 );
-
-    TEST_ASSERT( mbedtls_ecdsa_sign( &grp, &r, &s, &d, buf, sizeof( buf ),
-                                     &mbedtls_test_rnd_pseudo_rand,
-                                     &rnd_info ) == 0 );
-    TEST_ASSERT( mbedtls_ecdsa_verify( &grp, buf, sizeof( buf ), &Q, &r, &s ) == 0 );
+    while ( test_runs-- )
```

Comment:
[Having looked further, this comment is not applicable. But since I've already typed it, I'm leaving it for future reference as to what to do when you want to run multiple passes of something in a test case.] <s>The first run has `test_runs == 1` and the second run has `test_runs == 0`. This is a bit confusing. Also, test failure messages won't convey which test run is failing. Please use (0,1) for the test runs (I'll also accept (1,2)), using a clear loop `for( test_run = 0; test_run <= 1; test_run++ )`, or better, put the loop body (apart from the RNG preparation) into a function and call it twice. Additionally, call `mbedtls_test_set_step` just before the RNG preparation so that the step (a.k.a. "test run") is shown in failure messages.</s>